File: react/perimeter/src/components/Trail.js
import React from "react";
import Trailheads from "./Trailheads";

const Trail = ({ trail }) => {
  const { name, footTravelOnly, length, trailheads } = trail;
  return (
    <div>
      <h3>{name}</h3>
      <ul>
        {footTravelOnly && (
          <li>
            <b>Foot travel only</b>
          </li>
        )}
        <li>Length: {length} miles</li>
        {trailheads.length > 0 && <Trailheads trailheads={trailheads} />}
      </ul>
    </div>
  );
};

export default Trail;

File: lib/perimeter/trails/trail.ex
defmodule Perimeter.Trails.Trail do
  use Ecto.Schema
  import Ecto.Changeset

  schema "trails" do
    field :foot_travel_only, :boolean, default: false
    field :length, :float
    field :name, :string

    has_many :trailheads, Perimeter.Trails.Trailhead

    timestamps()
  end

  @doc false
  def changeset(trail, attrs) do
    trail
    |> cast(attrs, [:name, :length, :foot_travel_only])
    |> validate_required([:name, :length, :foot_travel_only])
  end
end
